### DISCUSSION TITLE
Step-by-step instructions for setting up Tesla Fleet API with DuckDNS, Nginx proxy and Tesla HTTP Proxy. Minimal prerequisites.

### DISCUSSION THREAD

**Discussion by jason-curtis** [alandtse/tesla#1048] [Show and tell]
I finally got my own setup working, so I figured I should share the steps that helped me get the Fleet API up and running on my setup. They're pretty rough step-by-step instructions, but they're more comprehensive than other instructions I've seen. I figured they might help some folks here (such as folks on the #976 thread).

~~Feel free to comment & suggest edits: [Google docs link]~~ EDIT: see below; there's a [Wiki](https://github.com/alandtse/tesla/wiki/Proxy-setup-with-DuckDNS-and-NGinx) page that I've contributed to rather than maintaining my own doc.

Also! Note that there's a brand-new [Tesla Fleet integration](https://www.home-assistant.io/integrations/tesla_fleet) that might replace the Tesla Custom Integration soon. For more, see discussion #1036 .

**Comment by alandtse** [comment]
Thanks for sharing. The [Wiki ](https://github.com/alandtse/tesla/wiki/Proxy-setup-with-DuckDNS-and-NGinx) is probably the best location for something like this. Dropping a link may work or just having the content there too so it's not tied to your google account.

**Comment by jason-curtis** [comment]
Oooh, that wiki is everything that I was looking for during this process, and it's more complete than my doc! Would you be open to linking to the wiki from more places like the repo README and [DOCS.md](https://github.com/llamafilm/tesla-http-proxy-addon/blob/main/tesla_http_proxy/DOCS.md)?

**Reply by alandtse** [reply]
PRs are always welcome.

**Reply by jason-curtis** [reply]
cool cool. submitted #1049 .

**Comment by Wintherman** [comment]
Hi, I have a Model-Y which I could read values from before Tesla stopped the easy reading. I could check the Batterylevel and start other activities when it reach the charge_limit etc.
But know I need to get the Tesla_HTTP_Proxy up and running and I am using your very good step-by-step instruction.
But I get stuck when the Tesla_HTTP_proxy start up and says "**Failed to start add-on**
Port '443' is already in use by something else on the host."  and I assume it is due to the NGINX run on port 443 since that is for HTTPS.
How can I solve this? 
If I try with port 8099 on the Tesla-HTTP_Proxy it starts but the logs does not look nice and after a restart of NGINX there is no https://[mydomain].duckdns.org/.well-known/appspecific/com.tesla.3p.public-key.pem to be found.

I am not sure if this is a bug so I post it here under this discussion.

One more thing, anyone now where in the HA file structure I can find ".well-known/appspecific/com.tesla.3p.public-key.pem"?
//Håkan

**Reply by Wintherman** [reply]
My problem solved by follow the step-by step very careful.
Then at Enrolling the key to the car only works if configuring the NGIX with a dummy domain first and then restart NGINX after after starting the Tesla HTTP Proxy and the start over the again with Tesla credential and restat the NGIX again. Mayby not exactly like this but it works after some restarts...

**Comment by kimme1024** [comment]
Just a follow up question:

My wife just bought a second hand model y with fleet api.  I still have owner api on my model s.
I want to use tesla solar charger in home assistant as I do with mine.

However it's a bit unclear regarding the http proxy installation.

I want to know if there are API limits using this solution? (I'm polling about every 35s and I'm sending contant changes for charing amps)
If there are limits, I won't need to waste time installing this and I will use the tessie/teslametry route instead.

Thanks in advance for answering this!

**Reply by skipishere** [reply]
Using the Fleet API you'll have the restrictions Tesla has set - https://developer.tesla.com/docs/fleet-api/billing-and-limits and the associated costs (after you've used up the 'free' allowance). Unless you're really needing the data every 35s I would strongly advise against polling it that often as that will stop your car from being able to go to sleep and will cause unnecessary battery drain.

In theory Tessie will have similar limits as it too is hitting the same API endpoints but I don't know what sort of caching they might be doing to help limit the costs they would incur

**Reply by kimme1024** [reply]
Thanks for replying so quickly.
I need the polling for charging with excess solar energy. This changes constantly. 
Is there another way without limits? Locally by example?

**Reply by skipishere** [reply]
Do you mean polling the solar generation / home battery? That currently doesn't have to use the Fleet API and you can get that data locally (for Tesla battery) as well

**Reply by kimme1024** [reply]
What is does exactly is the following:

Every 35s it checks my load balance (production-consumption). In case of excess solar the script changes the amperage of the car by 1 or more.
During winter I'm doing exactly the same, but the other way around, I set a max draw from grid and the script adapts the amperage of the car.

So indeed, the part of of calculating the load balance doesn't need the Fleet API, but constantly adapting the car to that value does.
